### ISSUE TITLE
subs=attributes in listing and specialchars escaping

### ISSUE THREAD

**Issue by oncletom** [asciidoctor/asciidoctor.js#413]
This is a follow-up of #408.

I replace the leading `$` of listings with an HTML tag. The $ sign is added back via CSS. Listings can be pasted in a terminal without pasting the leading `$`.

<img width="533" alt="image" src="https://user-images.githubusercontent.com/138627/35238792-e1ae0442-ffae-11e7-9606-34f57f1e8b87.png">

As you can see on the above screenshot, one example contains the escaped HTML chars. When `subs=attributes` is on, the escaping does not happen (and the `$` is well displayed). I have tried to alter the block `subs` (even with `setAttribute('subs', '-specialchars')`) but without any luck (in the `TreeProcessor#process()`).

I'm not quite sure how to replicate the behaviour of the listing with `subs=attribute` with the second example? I've seen a reference to `default_subs` in the block attributes… but no place to define this as a default document attribute.

**Comment by Mogztter** [comment]
I'm wondering if you could reproduce this issue in Asciidoctor (Ruby) to make sure that this isn't related to the JavaScript runtime ?

By default, the normal substitution should be applied: http://asciidoctor.org/docs/user-manual/#subs
I believe this behavior is intended because the block contains HTML, and the normal substitution is supposed to escape HTML tags.

> I have tried to alter the block subs

Did you try to specify the `subs` option in the document ?

> even with setAttribute('subs', '-specialchars')

`subs` is an option so as far as I know `setAttribute('subs', '-specialchars')` will do nothing.
Could you try to use `subs=none` ?

I think what you should do is remove `specialchars` with `remove_sub` when you are inserting HTML code in your block:

```js
if (isModified) {
  const blockIndex = index(parent.getBlocks(), block);
  // update lines
  block.lines = lines;
  // calling Opal function directly with $
  block.$remove_sub('specialchars');
  parent.blocks[blockIndex] = block;
}
```

Although I'm not sure that `specialchars` will be present in the `subs`, could you add a `console.log('block.subs', block.subs)` for debugging purpose ?

**Comment by oncletom** [comment]
I discovered the _Text Substitutions_ doc section yesterday. It has been super useful—thanks if this content is new :-)

Thanks for the `remove_sub` method! I forgot the Ruby API could be called directly.

I ended up writing using `specialchars`:

```js
block.$remove_sub('specialcharacters');
```

According to the doc, it's the default subs for listings.

Thanks for your help; you're so knowledgeable about it!
